xls/public/c_api_impl_helpers.cc

```diff
@@ -0,0 +1,88 @@
+// Copyright 2024 The XLS Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "xls/public/c_api_impl_helpers.h"
+
+#include <filesystem>
```

Comment:
`// NOLINT`?

Reply:
Done, though note it's hard to anticipate what the linter will do for contributors, I don't think it's externally available right?

Reply:
> Done, though note it's hard to anticipate what the linter will do for contributors, I don't think it's externally available right?

@hzeller was mentioning it would be nice to add the `NOLINT` annotation on import, or make sure we enforce the check externally (maybe enforcing `xls::common::Filesystem` usage?)